pkg/auto/exporthttp/job/job.go

```diff
@@ -28,15 +29,21 @@ var (
 	errNoSensorsRetrieved = errors.New("no sensors retrieved")
 )
 
+// boltKeyTemplate is the template string used to generate the bolt key
+// "${AutoName}_${config.Root.Name}_${Job.Name}"
+// It is used to store the job's last previous execution timestamp in the database.
+const boltKeyTemplate = "%s_%s_%s"
+
 const defaultTimeout = time.Second * 5
 
 // Job represents an exporthttp automation task that executes Do to send a POST request
 type Job interface {
 	GetName() string
 	GetUrl() string
 	GetSite() string
-	GetExecutionAfter(t time.Time) <-chan time.Time
-	SetPreviousExecution(t time.Time)
+	GetExecutionAfter(autoName, scName, jobName string) <-chan time.Time
+	SetPreviousExecution(autoName, scName, jobName string, t time.Time)
+	GetSchedule() *jsontypes.Schedule
```

Comment:
Why pass all these args into the func? Can't the job just know these things? The job is created once during the config processing after all, so it knows these at that time.